### ISSUE TITLE
Bug in receive bytes_function

### ISSUE THREAD

**Issue by gorankecman** [IRNAS/Koruza-Move-Driver-Firmware#29]
Function receive_bytes occasionally presented a bug where for certain frames received frame was shortened or zeros have been returned as a final result. This occurs when sending move commands to stepper motors and usually when the frame contains special frame markers (escape sequence for example). 
One example of a wrongly interpreted frame is: 
F1 01 00 01 02 04 00 0C 00 00 00 66 FF FF E9 44 00 00 00 00 03 00 04 68 14 F3 F3 F3 F2 F2

**Comment by gorankecman** [comment]
In order to find the root cause of this bug, the code was analyzed in details. The problem was caused by handling escape sequence which is used to avoid misinterpreting special frame markers for frame start (0xF1), frame stop (0xF2) and frame marker escape (0xF3) as regular message content.
The bug was identified in part of the code which handles frame stop condition:

`/* End byte received */
    else if (rx_data[0] == FRAME_MARKER_END)
    {
      /* End byte received in the frame */
      if ((rx_last[0] == FRAME_MARKER_ESCAPE) && (rx_last[1] != FRAME_MARKER_ESCAPE) && (rx_buffer[0] == FRAME_MARKER_START))
      {
        rx_buffer[rx_indx++] = rx_data[0];
      }
      /* Real end byte received */
      else if ((rx_last[0] != FRAME_MARKER_ESCAPE || (rx_last[0] == FRAME_MARKER_ESCAPE && rx_last[1] == FRAME_MARKER_ESCAPE)) && rx_buffer[0] == FRAME_MARKER_START)
      {
        rx_buffer[rx_indx++] = rx_data[0];
        message_len = rx_indx;
        rx_indx = 0;
        rx_last[0] = 0;
        rx_last[1] = 0;
        /* Transfer complete, data is ready to read */
        command_received = true;
        /* Disable USART1 interrupt */
        //HAL_NVIC_DisableIRQ(USART1_IRQn);
      }
    }`

In general, the problem was in the case when there were more than two escape (0xF3) characters in line. The code was limited to checking only two escape characters in a row (this is especially the problem when it comes to the end of frame and command 0xF2 (end frame) should be received). This can be seen from the condition:

`else if ((rx_last[0] != FRAME_MARKER_ESCAPE || (rx_last[0] == FRAME_MARKER_ESCAPE && rx_last[1] == FRAME_MARKER_ESCAPE)) && rx_buffer[0] == FRAME_MARKER_START)`

In order to fix that, we have added tracking of the escape characters throughout whole execution of this function. If one escape character occurs, we have marked that event, and then after next character is received (either 0xF3 again or 0xF1 (start frame) or 0xF2 (end frame)) corresponding action was taken and escape event flag was reset. In this way, it cannot happen that any escape character is misinterpreted as it was the case before.

Fix of this bug can be found on develop branch, commit ID: 
`bfe9699a6570fcaf974314c5607b5229a96d428c`
